polaris/benchmark/_base.py

```diff
@@ -188,153 +180,54 @@ def _validate_main_metric_is_in_metrics(self) -> Self:
         return self
 
     @model_validator(mode="after")
-    def _validate_split(self) -> Self:
+    def _validate_cols(self) -> Self:
         """
-        Verifies that:
-          1) There are no empty test partitions
-          2) All indices are valid given the dataset
-          3) There is no duplicate indices in any of the sets
-          4) There is no overlap between the train and test set
-          5) No row exists in the test set where all labels are missing/empty
+        Verifies that all specified columns are present in the dataset.
         """
+        columns = self.target_cols | self.input_cols
+        dataset_columns = set(self.dataset.columns)
+        if not columns.issubset(dataset_columns):
+            raise InvalidBenchmarkError("Not all specified columns were found in the dataset.")
 
-        if not isinstance(self.split[1], dict):
-            self.split = self.split[0], {"test": self.split[1]}
-        split = self.split
-
-        # Train partition can be empty (zero-shot)
-        # Test partitions cannot be empty
-        if any(len(v) == 0 for v in split[1].values()):
-            raise InvalidBenchmarkError("The predefined split contains empty test partitions")
+        return self
 
-        train_idx_list = split[0]
-        full_test_idx_list = list(chain.from_iterable(split[1].values()))
+    @field_serializer("metrics")
+    def _serialize_metrics(self, value: set[Metric]) -> list[Metric]:
+        """
+        Convert the set to a list. Since metrics are models and will be converted to dict,
+        they will not be hashable members of a set.
+        """
+        return list(value)
```

Comment:
**Comment:** This is outside the scope of this PR, but I believe we may not be using Pydantic to its full potential here. Pydantic differentiates two `mode`s for its [`model_dump`](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_dump), i.e. `json` and `python`. I wonder if this conversion could be done automatically if we specify the `json` mode? And more generally, the use of these two modes seems like something we can be more rigorous about.

Reply:
Serialization works correctly with `mode="json"`, even with this omitted. The thing is, these are still Pydantic models, so we should make sure Python mode for `model_dump` also works. This feels like a good middle ground.